example/main.go

```diff
@@ -75,6 +82,55 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Setup webhooks
+	mutatingWebhook, err := builder.NewWebhookBuilder().
```

Comment:
IMO we want to hide some complexity from the webhook developer.
E.g. most users don't need to understand how to use [RuleWithOperations API](https://github.com/kubernetes/api/blob/f5c295feaba2cbc946f0bbb8b535fc5f6a0345ee/admissionregistration/v1beta1/types.go#L136-L142). Instead they can simply call `Operations()` and `ForType()`, we will set the `RuleWithOperations` for them.

For advanced users who know what they are doing, they can directly use the `Webhook` struct instead of go through the `Builder`.